### PR TITLE
file,server: add method for parsing JSON with custom validation

### LINKED ISSUE #944: http: support validation options when creating a file through JSON
ACH Version: `v1.9.2`

**What were you trying to do?**
Create an ACH file from JSON through the HTTP server with validation options set.

**What did you expect to see?**
A successful request given the validation options.

### PR DESCRIPTION
**Pull request by adamdecaf**
Fixes: https://github.com/moov-io/ach/issues/944

### CODE REVIEW

File: file_test.go [hunk 1]
```diff
@@ -30,6 +30,8 @@ import (
 	"time"
 
 	"github.com/moov-io/base"
+
+	"github.com/stretchr/testify/require"
```
Comment (vxio) [inline file_test.go:34]: Sweet, I was hesitant about using `testify` in this repo, but I'll start using it too

File: test/testdata/json-bypass-destination.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
+{
+    "id": "adam-01",
+    "fileHeader": {
+        "id": "adam-01",
+        "immediateDestination": "000000000",
+        "immediateOrigin": "000000000",
```
Comment (vxio) [inline test/testdata/json-bypass-destination.json]: I recommend renaming this file `json-bypass-origin-and-destination.json`

File: file_test.go [hunk 3]
```diff
@@ -644,6 +646,39 @@ func TestFile__IATdatetimeParse(t *testing.T) {
...
+	require.NoError(t, err)
+	require.NoError(t, file.Validate())
+
+	require.Equal(t, "adam-01", file.ID)
+	require.Equal(t, "000000000", file.Header.ImmediateOrigin)
+	require.Equal(t, "231380104", file.Header.ImmediateDestination)
+	require.Equal(t, "181008", file.Header.FileCreationDate)
+}
+
+func TestFile__JsonBypassDestination(t *testing.T) {
```
Comment (vxio) [inline file_test.go]: I recommend renaming this test `..__JsonBypassOriginAndDestination`

**Review by vxio — DISMISSED** [review]
Looks good! Just had some minor naming suggestions.

File: file_test.go (hunk 1 above)
Comment (adamdecaf) [reply file_test.go:34]: Yea. I figure it's good enough. It'll only be included in tests also.

**Review by vxio — APPROVED** [review]

**Pull request merged**